clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -464,14 +464,6 @@ public Node provide() {
          * @param throwable     The failure exception.
          */
         final void fail(long now, Throwable throwable) {
-            if (aborted) {
```

Comment:
Moving this below the `UnsupportedVersionException` check means that we may retry even though the call has been aborted. Doesn't seem like we'd want to do that or am I missing something?

Reply:
Currently, we should not be able to get an `UnsupportedVersionException` in combination with `aborted` = true.  However, I agree that in general, it would be conceptually clearer to check `aborted` first.  So I'll move it back up to the top.